### DISCUSSION TITLE
Assigning Categories to Item

### DISCUSSION THREAD

**Discussion by rogercreagh** [TechbeeAT/jtxBoard#847] [Ideas]
@patrickunterwegs  I'm really liking the new selection method for categories in the filter dialog in 2.5.5beta. Thank you for a big improvement.

I wonder what you'd think about implementing the same method for selecting categories to assign to an item. Currently you get a number of most recently used categories in a visible horizontal list and a text box for typing a new or existing category name, which is okay if you can remember the name of the category you want. If you want to see the list of existing categories you have to scroll the visible list horizontally and they appear in most recent order. This is awkward to use for selecting multiple categories.

Two improvements - 
1. rather than a horizontally scrolling list do it like the filter selection with a 'more' button which reveals the entire list sorted alphabetically - far quicker and more convenient if you want to assign several categories, or if you can't remember if your category for visiting people is called 'trips' or 'visits' for example. 
2. the text entry box is worth keeping in addition to the filter style selection, but the search is both case sensitive and not case sensitive which is confusing. It defaults to starting your text with a capital letter but the search and matching list is case insensitive, but if you have a category called 'visits', say, and you quickly type 'Visits' and press enter it will create a new category called 'Visits' and assign that - so now you have both 'visits' and 'Visits'

Personally I always use lower case for category names unless they are proper names which I do capitalize. So when creating a new category on the fly I usually have to start by remembering to turn caps off. 

Also in other apps I am used to using category and tag names are always treated case insensitive so 'Steve' would match 'steve' and if tag 'Steve' already exists searching for 'steve' would bring up only 'Steve' and trying to create a new category 'steve' would raise a warning 'Category Steve already exists' (or in the case of searching to select a tag it would simply select 'Steve' and not create a new potential duplicate). 
To me this seems more sensible - but others may have different views?

**Comment by bepolymathe** [comment]
Hi, 

> Also in other apps I am used to using category and tag names are always treated case insensitive so 'Steve' would match 'steve' and if tag 'Steve' already exists searching for 'steve' would bring up only 'Steve' and trying to create a new category 'steve' would raise a warning 'Category Steve already exists' (or in the case of searching to select a tag it would simply select 'Steve' and not create a new potential duplicate).
> To me this seems more sensible - but others may have different views?

I also work like this, but I think case sensitivity might be an option.

**Comment by patrickunterwegs** [comment]
Hey, sorry for the late reply, I have a new job and times are a bit busy.

So about your proposals: 

> rather than a horizontally scrolling list do it like the filter selection with a 'more' button which reveals the entire list sorted alphabetically - far quicker and more convenient if you want to assign several categories, or if you can't remember if your category for visiting people is called 'trips' or 'visits' for example.

I am open for a change there, also to make it consistent with the filter selection. 
But you mean here to keep the horizontal scroll, limit to like 5 entries, then a more button, right? This would also mean that suddenly all categories appear which could fill up the screen... couldn't that end up being too much?


> the text entry box is worth keeping in addition to the filter style selection, but the search is both case sensitive and not case sensitive which is confusing. It defaults to starting your text with a capital letter but the search and matching list is case insensitive, but if you have a category called 'visits', say, and you quickly type 'Visits' and press enter it will create a new category called 'Visits' and assign that - so now you have both 'visits' and 'Visits'

I get it, it definitely makes sense. So here I would suggest: When pushing enter (or the icon button at the end of the edit text), the app should check case-insensitive if the given category already exists. If yes, take the first match (just taking whatever cases it has), if no, insert a new one. It might still take the wrong category (with other cases) unless there is a unique match. 
What do you think?

**Comment by rogercreagh** [comment]
> But you mean here to keep the horizontal scroll, limit to like 5 entries, then a more button, right? This would also mean that suddenly all categories appear which could fill up the screen... couldn't that end up being too much?

Yes that's exactly what I mean. They might indeed fill the screen, which is the whole point if you have a lot of cats and want to see what you've got, or select several at once. 
It would need an enhancement which is missing on the filter screen (but could usefully be added there as well) and that is when the full list is visible replace the `more` button with `less` which would collapse the list back to the otiginal howevermamy.

**Reply by patrickunterwegs** [reply]
Sounds good! I'll do it in #866

**Comment by rogercreagh** [comment]
> I get it, it definitely makes sense. So here I would suggest: When pushing enter (or the icon button at the end of the edit text), the app should check case-insensitive if the given category already exists. If yes, take the first match (just taking whatever cases it has), if no, insert a new one. It might still take the wrong category (with other cases) unless there is a unique match.

The ideal would be three steps
 1 take a case match if there is one
 2 if no case match take a case insentive match if there is one
 3 if no case insensitive match create a new one

But obvs what you suggest is simpler (maybe only a bit?) to implement and would be an improvement - but i would probably file a long term request to take it to the three levels one day 😁

**Reply by patrickunterwegs** [reply]
Thanks, yes, your solution sounds more complete 😉  I'll do it in #865

**Reply by patrickunterwegs** [reply]
Step 2 will still require an additional condition, so if no case match take THE FIRST case insentive match if there is one 😉
